src/main/java/ca/on/oicr/pineryreports/reports/impl/OctaneCountsReport.java

```diff
@@ -0,0 +1,537 @@
+            AttributeDto remainingAttr = parent.getAttributes().stream().filter(attr -> ATTR_REMAINING.equals(attr.getName())).findFirst().get();
+            Integer remaining = Integer.parseInt(remainingAttr.getValue());
+            Integer consumed = Integer.parseInt(getAttribute(ATTR_CONSUMED, sam));
+            remainingAttr.setValue(Integer.toString(remaining - consumed));
+        }));
+  }
+  
+  private int countSlidesRemaining(Collection<SampleDto> samples) {
+    return samples.stream()
+        .mapToInt(s -> getIntAttribute(ATTR_REMAINING, s))
+        .sum();
+  }
+  
+  private int countSlidesReceived(Collection<SampleDto> samples) {
+    return samples.stream()
+        .mapToInt(s -> getIntAttribute(ATTR_SLIDES, s))
+        .sum();
+  }
+  
+  private static Predicate<SampleDto> byStain(String stain) {
+    return slide -> stain == null ? getAttribute(ATTR_STAIN, slide) == null : stain.equals(getAttribute(ATTR_STAIN, slide));
+  }
+  
+  private static List<SampleDto> filterNonEmpty(Collection<SampleDto> samples) {
+    return samples.stream()
+        .filter(sample -> !"EMPTY".equals(sample.getStorageLocation()))
+        .collect(Collectors.toList());
+  }
+  
+  private final Predicate<SampleDto> withSlidesRemaining = slide -> {
+    Integer slides = getIntAttribute(ATTR_SLIDES, slide);
+    Integer discards = getIntAttribute(ATTR_DISCARDS, slide);
+    if (slides == null) {
+      throw new IllegalArgumentException("Sample does not seem to be a slide");
+    }
+    if (discards == null) {
+      return slides > 0;
+    }
+    return slides > discards;
+  };
+  
+  /**
+   * Find the number of unique identities amongst a set of new samples, excluding those with descendants created before the
+   * report start date in another set of samples. For example, this can be used to find the number of Identities for which slides
+   * were received for the first time during the reporting period. To determine this, newSamples would contain all slides created
+   * during the reporting period, and potentialOldSamples would contain all slides ever created
+   * 
+   * @param newSamples filtered samples for initial Identity count
+   * @param potentialOldSamples samples to exclude Identities for if the sample's creation date is before the report start date
+   * @param potentialParents all samples, used for navigating the hierarchy from the other samples to their Identities
+   * @return the count of unique Identities with descendants in newSamples, but no descendants created before the report start
+   * date in potentialOldSamples
+   */
+  private long countNewCases(Collection<SampleDto> newSamples, Collection<SampleDto> potentialOldSamples,
+      Map<String, SampleDto> potentialParents) {
+    return getUniqueIdentities(newSamples, potentialParents).stream()
+        .filter(identity -> potentialOldSamples.stream()
+        .noneMatch(sam -> identity.getId().equals(getParent(sam, "Identity", potentialParents).getId())
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'SAMPLE_CATEGORY_IDENTITY' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)